src/pages/projectSettings/tabs/EventLogTab/EventLogDiffs.ts

```diff
@@ -0,0 +1,55 @@
+import { diff } from "deep-object-diff";
+
+const getDiffProperties = (eventObj: object) => {
+  const isObject = (val) =>
+    val && typeof val === "object" && !Array.isArray(val);
+
+  const addDelimiter = (a, b) => (a ? `${a}.${b}` : b);
+
+  const paths = (obj = {}, head = "") =>
+    Object.entries(obj).reduce((event, [key, value]) => {
+      const fullPath = addDelimiter(head, key);
+
+      return isObject(value)
+        ? event.concat(paths(value, fullPath))
+        : event.concat(fullPath);
+    }, []);
+
+  return paths(eventObj);
+};
+
+const getNestedObject = (nestedObj: object, pathArr: []) =>
+  pathArr.reduce(
+    (obj, key) => (obj && obj[key] !== "undefined" ? obj[key] : undefined),
+    nestedObj
+  );
+
+type EventDiffLine = {
+  key: string;
+  before: any;
+  after: any;
+};
+
+export const getEventDiffLines = (
+  before: object,
+  after: object
+): EventDiffLine[] => {
+  const eventDiff = diff(before, after);
+  const pathKeys = getDiffProperties(diff(before, after));
```

Comment:
No need to call `diff()` again
```suggestion
  const pathKeys = getDiffProperties(eventDiff);
```

Reply:
oops, I refactored and forgot to update. thanks!